code/app/com/feth/play/module/pa/providers/oauth2/eventbrite/EventBriteAuthProvider.java

```diff
@@ -36,13 +38,8 @@ protected EventBriteAuthUser transform(final EventBriteAuthInfo info, final Stri
 
         final String url = getConfiguration().getString(
                 USER_INFO_URL_SETTING_KEY);
-        final WSResponse r = WS
-                .url(url)
-                .setQueryParameter(TOKEN,
-                        info.getAccessToken())
-                .get()
-                .get(getTimeout());
 
+        final WSResponse r = fetchAuthResponse(url, new QueryParam(TOKEN, info.getAccessToken()));
         final JsonNode result = r.asJson();
         if (r.getStatus() >= 400) {
             throw new AuthException(result.get("meta").get("errorDetail").asText());
```

Comment:
This class and others that provide a concrete implementation of the AuthProvider class should be annotated with the javax.inject.Singleton annotation, because they depend on the ApplicationLifecycle class. see [https://www.playframework.com/documentation/2.5.x/JavaDependencyInjection#Stopping/cleaning-up]